include/fast_float/parse_number.h

```diff
@@ -60,6 +60,53 @@ from_chars_result parse_infnan(const char *first, const char *last, T &value)  n
   return answer;
 }
 
+/**
+ * Returns true if the floating-pointing rounding mode is to 'nearest'.
+ * It is the default on most system. This function is meant to be inexpensive.
+ * Credit : @mwalcott3
+ */
+fastfloat_really_inline bool rounds_to_nearest() noexcept {
+  // See
+  // A fast function to check your floating-point rounding mode
+  // https://lemire.me/blog/2022/11/16/a-fast-function-to-check-your-floating-point-rounding-mode/
+  //
+  // This function is meant to be equivalent to :
+  // prior: #include <cfenv>
+  //  return fegetround() == FE_TONEAREST;
+  // However, it is expected to be much faster than the fegetround()
+  // function call.
+  //
+  // The volatile keywoard prevents the compiler from computing the function
+  // at compile-time.
+  // There might be other ways to prevent compile-time optimizations (e.g., asm).
+  // The value does not need to be std::numeric_limits<float>::min(), any small
+  // value so that 1 + x should round to 1 would do.
```

Comment:
This actually isn't true.  The problem is with excess precision.  E.g. on i?86 32-bit (or x86-64 with -mfpmath=387) floats are evaluated to the precision of long double (with -fexcess-precision=standard) or could be either way (otherwise).
With std::numeric_limits<float>::min() it will work in this case, because even i?86 long double has just 64-bit mantissa and because std::numeric_limits<float>::min() is 2^-126, both will still round to nearest to 1.0.
But if the value wasn't that small, but say just 2^-60, it wouldn't be true anymore.  Fortunately it would then return false and so result in the safer version despite perhaps being really FE_TONEAREST.
2^-126 is good even if some target hypothetically evaluates everything in IEEE quad (113 bit mantissa) and no hw AFAIK implements 256-bit floats right now.

Reply:
Point taken. I have added a remark in a follow-up commit which states "after accounting for excess precision".